pkg/provisioning/changelog.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provisioning
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+// changelog should be adjusted every time we change the pipeline config and add
+// a new config version. Based on the changelog the parser will output warnings.
+var changelog = map[string][]change{
+	"1.0": {}, // no changes, initial version
+	"1.1": {{
+		field:      "pipelines.*.dead-letter-queue",
+		changeType: fieldIntroduced,
+		message:    "field dead-letter-queue was introduced in version 1.1, please update the pipeline config version",
+	}},
+}
+
+// change is a single change that was introduced in a specific version.
+type change struct {
+	// field contains the path to the field that was changed. Nested fields can
+	// be represented with dots (e.g. nested.field)
+	field      string
+	changeType changeType
+	// message is the log message that will be printed if a file is detected
+	// that uses this field with an unsupported version.
+	message string
+}
+
+// changeType defines the type of the change introduced in a specific version.
+type changeType int
+
+const (
+	fieldDeprecated changeType = iota
+	fieldIntroduced
+)
+
+// expandedChangelog is populated in init based on changelog. This map contains
+// the changelog in a structure that is useful for traversing in configLinter.
```

Comment:
Maybe mention an example of how it should look like? I think it would help understand the code better.

Reply:
I should probably write a test for this, I somehow missed it. That's at least what I would write in a review comment if I saw this 😄 I hope this will count as providing an example.